src/zc/buildout/easy_install.py

```diff
@@ -1658,7 +1666,117 @@ def call_easy_install(spec, dest):
             "Look above this message for any errors that "
             "were output by easy_install.",
             spec)
-    return glob.glob(os.path.join(dest, '*'))
+
+    split_entries = [os.path.splitext(entry) for entry in os.listdir(dest)]
+    try:
+        distinfo_dir = [
+            base + ext for base, ext in split_entries if ext == ".dist-info"
+        ][0]
+    except IndexError:
+        logger.error(
+            "No .dist-info directory after installing %s",
+            spec)
+        raise
+
+    return make_egg_after_pip_install(dest, distinfo_dir)
+
+
+def make_egg_after_pip_install(dest, distinfo_dir):
+    """build properly named egg directory"""
+
+    # `pip install` does not build the namespace aware __init__.py files
+    # but they are needed in egg directories.
+    # Add them before moving files setup by pip
+    ns_file = os.path.join(dest, distinfo_dir, 'namespace_packages.txt')
+    if os.path.exists(ns_file):
+        with open(ns_file) as f:
+            namespace_packages = [
+                line.strip().replace('.', os.path.sep)
+                for line in f.readlines()
+            ]
+
+        for namespace_package in namespace_packages:
+            init_py_file = os.path.join(dest, namespace_package, '__init__.py')
+            with open(init_py_file, 'w') as f:
+                f.write(
+                    "__import__('pkg_resources').declare_namespace(__name__)"
+                )
+
+    # Remove `bin` directory if needed
+    # as there is no way to avoid script installation
+    # when running `pip install`
+    ep_file = os.path.join(dest, distinfo_dir, 'entry_points.txt')
+    if os.path.exists(ep_file):
+        with open(ep_file) as f:
+            if "console_scripts" in f.read():
+                bin_dir = os.path.join(dest, 'bin')
```

Comment:
I'm now thinking about Windows and how the virtualenv `bin` directory is named `Scripts` there.